src/apps/search-result/search-result.test.ts

```diff
@@ -0,0 +1,127 @@
+const coverUrlPattern = /^https:\/\/res\.cloudinary\.com\/.*\.(jpg|jpeg|png)$/;
+
+describe("Search Result", () => {
+  it("Should render the search results as expected", () => {
+    cy.visit(
+      "/iframe.html?id=apps-search-result--search-result&args=pageSizeDesktop:2;pageSizeMobile:2"
+    );
+
+    cy.log("Check length of search result list:");
+    cy.get(".search-result-page__list").find("li").should("have.length", 2);
+
+    cy.log("Do the search results have images?");
+    cy.get(".search-result-page__list .search-result-item img")
+      .should("have.attr", "src")
+      .and("match", coverUrlPattern);
+
+    cy.log("Does the search result have favourite buttons?");
+    cy.get(
+      ".search-result-page__list .search-result-item .button-favourite"
+    ).should("have.attr", "aria-label", "Add to favorites");
+
+    cy.log("Does the search result have a series line?");
+    cy.get(
+      ".search-result-page__list .search-result-item .horizontal-term-line"
+    ).should(
+      "contain.text",
+      "Nr. 1  i serienDummy Some SeriesNr. 1  i serienDummy Some Series"
+    );
+
+    cy.log("Does the search result have titles?");
+    cy.get(".search-result-page__list .search-result-item h2").should(
+      "contain.text",
+      "Dummy Some Title: Full"
+    );
+
+    cy.log("Does the search result have authors?");
+    cy.get(
+      ".search-result-page__list .search-result-item p.text-small-caption"
+    ).should("contain.text", "Af Dummy Jens Jensen (Dummy 1839)");
```

Comment:
Consider reducing the dependency on CSS selectors.

Cypress and others recommend testing based on what is visible to end users or assistive technologies to reduce frailty. In this regard it is sufficient that the UI contains the text "Af Dummy Jens Jensen (Dummy 1839)".

This also applies for other tests in this PR.

Reply:
Check